pkg/provisioner/ironic/ironic.go

```diff
@@ -1011,8 +1020,21 @@ func (p *ironicProvisioner) Provision(getUserData provisioner.UserDataSource) (r
 		return p.startProvisioning(ironicNode, checksum, getUserData)
 
 	case nodes.Manageable:
+		if len(cleanSteps) > 0 && len(ironicNode.CleanStep) == 0 {
+			return p.startManualCleaning(ironicNode, cleanSteps)
+		}
 		return p.startProvisioning(ironicNode, checksum, getUserData)
 
+	case nodes.CleanFail:
+		if ironicNode.LastError == "" {
+			p.log.Info("failed but error message not available")
+			result.Dirty = true
+			return result, nil
+		}
+		p.log.Info("found error", "msg", ironicNode.LastError)
+		result.ErrorMessage = fmt.Sprintf("Hardware configuration failed")
```

Comment:
This error message should include ironicNode.LastError.

Reply:
Done :)